admin/horace_install.m

```diff
@@ -1,40 +1,41 @@
 function horace_install()
 %HORACE_INSTALL install this instance of Horace
 %
-HORACE_ON_SUB_PATTERN = '${Horace_CORE}';
-HERBERT_ON_SUB_PATTERN = '${Herbert_CORE}';
+HORACE_ON_PLACEHOLDER = '${Horace_CORE}';
+HERBERT_ON_PLACEHOLDER = '${Herbert_CORE}';
 
 install_root = fileparts(mfilename('fullpath'));
 
 % Find directories containing herbert/horace_init.m
-hor_init_dir = fileparts( ...
-    find_file('horace_init.m', {fullfile(install_root, 'Horace')}) ...
-);
-her_init_dir = fileparts( ...
-    find_file('herbert_init.m', {fullfile(install_root, 'Herbert')}) ...
-);
-
-% Find and update horace_on
-horace_on_path = find_file('horace_on.m.template', {install_root});
-horace_on_contents = fileread(horace_on_path);
-new_hor_on_contents = replace(horace_on_contents, HORACE_ON_SUB_PATTERN, hor_init_dir);
-new_hor_on_contents = replace(new_hor_on_contents, HERBERT_ON_SUB_PATTERN, her_init_dir);
-
-% Find and update herbert_on
-herbert_on_path = find_file('herbert_on.m.template', {install_root});
-herbert_on_contents = fileread(herbert_on_path);
-new_her_on_contents = replace(herbert_on_contents, HERBERT_ON_SUB_PATTERN, her_init_dir);
+% Do this first before find_userpath, so we can error out before potentially
+% creating a directory
+hor_init_dir = find_directory('horace_init.m', {fullfile(install_root, 'Horace')});
+her_init_dir = find_directory('herbert_init.m', {fullfile(install_root, 'Herbert')});
 
 % Find/create userpath - this is automatically added to Matlab path on launch
 user_path = find_userpath();
 
-% Write the herbert/horace_on files (containing paths to package) into userpath
-write_file(fullfile(user_path, 'horace_on.m'), new_hor_on_contents);
-write_file(fullfile(user_path, 'herbert_on.m'), new_her_on_contents);
+% Find and install horace_on
+horace_on_path = find_file('horace_on.m.template', {install_root});
+install_file( ...
+    horace_on_path, ...
+    fullfile(user_path, 'horace_on.m'), ...
+    {HORACE_ON_PLACEHOLDER, HERBERT_ON_PLACEHOLDER}, ...
+    {hor_init_dir, her_init_dir} ...
+);
+
+% Find and install herbert_on
+herbert_on_path = find_file('herbert_on.m.template', {install_root});
+install_file( ...
+    herbert_on_path, ...
+    fullfile(user_path, 'herbert_on.m'), ...
+    {HERBERT_ON_PLACEHOLDER}, ...
+    {her_init_dir} ...
+);
 
 % Copy worker_v2 script (required by parallel routines) to userpath
 worker_path = find_file('worker_v2.m.template', {install_root});
-copy_file(worker_path, fullfile(user_path, 'worker_v2.m'));
+install_file(worker_path, fullfile(user_path, 'worker_v2.m'));
```

Comment:
does redirecting this `copy()` though `install_file`  complicate things? It changes that from being a simple function into one with some logic